Vikta-API-testing-sample/src/test/java/com/test/api/address/AddressTest.java

```diff
@@ -0,0 +1,412 @@
+
+        AddressDTO createdAddress = given().spec(defaultRequestSpec())
+                .body(newAddress)
+                .log().all()
+                .when()
+                .post(getTestEnvironment().getAddressPath())
+                .then()
+                .log().all()
+                .spec(defaultResponseSpec())
+                .extract().as(AddressDTO.class);
+
+        long id = createdAddress.getId();
+
+        Response responseDelete = given().spec(defaultRequestSpec())
+                .queryParam(idQueryParam, id)
+                .when()
+                .log().all()
+                .delete(getTestEnvironment().getAddressPath())
+                .then()
+                .log().all()
+                .spec(defaultResponseSpec())
+                .extract().response();
+
+        Response responseGetDeleted = getResponseFromGetAddressById(id);
+
+        assertThat(responseDelete.getStatusCode()).isEqualTo(200);
+
+        assertThat(responseGetDeleted.getStatusCode())
+                .as("Address should be deleted")
+                .isEqualTo(404);
+    }
+
+    @Test
+    public void canNotDeleteNonExistingAddress(){
+        int id = 0;
+
+        Response response = getResponseFromGetAddressById(id);
+
+        assertThat(response.getStatusCode()).isEqualTo(404);
+    }
+
+    /**
+     * Methods returns List of Addresses
+     *
+     * @return List<AddressDTO>
+     */
+    private List<AddressDTO> getAddressesList() {
+        return given().spec(defaultRequestSpec())
+                .when()
+                .get(getTestEnvironment().getAddressListPath())
+                .then()
+                .spec(defaultResponseSpec())
+                .extract().jsonPath().getList(".", AddressDTO.class);
+    }
+
+    /**
+     * Methods returns AddressDTO object and expects that Address exist
+     *
+     * @param id address identifier
+     * @return AddressDTO object
+     */
+    private AddressDTO getAddressById(long id) {
+        return given().spec(defaultRequestSpec())
+                .queryParam(idQueryParam, id)
+                .when()
+                .get(getTestEnvironment().getAddressPath())
+                .then()
+                .spec(defaultResponseSpec())
+                .extract().as(AddressDTO.class);
+    }
+
+    /**
+     * Methods returns response from get request and expects that
+     * Address with given id exist
+     *
+     * @param id address identifier
+     * @return Response object
+     */
+    private Response getResponseFromGetAddressById(long id) {
+        return given().spec(defaultRequestSpec())
+                .queryParam(idQueryParam, id)
+                .when()
+                .get(getTestEnvironment().getAddressPath())
+                .then()
+                .spec(defaultResponseSpec())
+                .extract().response();
+    }
+
+    /**
+     * Method returns id existing user
+     *
+     * @return long id
+     */
+    private long getExistingUserId(){
```

Comment:
I think I saw this method in different classes too. If that's the case, please move it to BaseTest class and mark as protected.